common/runtime/src/test/java/io/quarkiverse/googlecloudservices/common/GcpDefaultsConfigSourceFactoryTest.java

```diff
@@ -0,0 +1,59 @@
+package io.quarkiverse.googlecloudservices.common;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.smallrye.config.ConfigSourceContext;
+import io.smallrye.config.ConfigValue;
+
+class GcpDefaultsConfigSourceFactoryTest {
+    @Test
+    void configSourceWorks() {
+        ConfigSourceContext context = Mockito.mock(ConfigSourceContext.class);
+        Mockito.when(context.getValue("quarkus.google.cloud.enable-metadata-server"))
+                .thenReturn(ConfigValue.builder().withValue("true").build());
+
+        Iterable<ConfigSource> configSources = new GcpDefaultsConfigSourceFactory(() -> "test-project-id")
+                .getConfigSources(context);
+        assertThat(configSources).asList().hasSize(1);
+
+        ConfigSource configSource = configSources.iterator().next();
+        assertThat(configSource.getProperties()).containsEntry("quarkus.google.cloud.project-id", "test-project-id");
+    }
+
+    @Test
+    void metadataServerDisabled() {
+        ConfigSourceContext context = Mockito.mock(ConfigSourceContext.class);
+        Mockito.when(context.getValue("quarkus.google.cloud.enable-metadata-server"))
+                .thenReturn(ConfigValue.builder().withValue("false").build());
+
+        Iterable<ConfigSource> configSources = new GcpDefaultsConfigSourceFactory(() -> "test-project-id")
+                .getConfigSources(context);
+        assertThat(configSources).isEmpty();
+    }
+
+    @Test
+    void staticOpenCensusOpenTelemetryInit() {
```

Comment:
What is the purpose of this test? Does it crash without the change (so it's a non-regression test?)

Reply:
Yes, fails without the prod code change. Added a comment.